src/core/quantum_gate.jl

```diff
@@ -305,7 +417,7 @@ julia> print(ψ_0)
 
 ```
 """
-function apply_gate!(state::Ket, gate::AbstractGate)
+function apply_gate!(state::Ket, gate::AbstractGateSymbol)
```

Comment:
this only works for `gate::Gate`, not if gate is a raw `AbstractGateSymbol` with no placement info